common/src/main/java/com/microsoft/identity/common/internal/eststelemetry/EstsTelemetry.java

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.eststelemetry;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.exception.BaseException;
+import com.microsoft.identity.common.internal.cache.ISharedPreferencesFileManager;
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager;
+import com.microsoft.identity.common.internal.logging.DiagnosticContext;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.result.AcquireTokenResult;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class EstsTelemetry {
+    private final static String TAG = EstsTelemetry.class.getSimpleName();
+
+    /**
+     * The name of the SharedPreferences file on disk for the last request.
+     */
+    private static final String LAST_REQUEST_TELEMETRY_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.last_request_telemetry";
+
+    private static volatile EstsTelemetry sEstsTelemetryInstance = null;
+    private static IRequestTelemetryCache sLastRequestTelemetryCache;
```

Comment:
Let's make `sEstsTelemetryInstance` and `sLastRequestTelemetryCache` _member_ variables of `EstsTelemetry` and keep a reference to a single instance -- rather than provide the _veneer_ of OO over a `static` implementation, my preference would be to have a _real_ OO implementation with an API that programmatically enforces that the entire app has a single instance

Reply:
Similar to many `Singleton` examples: https://www.baeldung.com/java-singleton

Reply:
Addressed